Framework/API/src/ISISInstrDataCache.cpp

```diff
@@ -0,0 +1,69 @@
+
+#include "MantidAPI/ISISInstrDataCache.h"
+#include "MantidAPI/FileFinder.h"
+#include "MantidKernel/ConfigService.h"
+#include "MantidKernel/Exception.h"
+#include "MantidKernel/InstrumentInfo.h"
+#include "MantidKernel/Logger.h"
+#include <algorithm>
+#include <boost/lexical_cast.hpp>
+#include <fstream>
+#include <iterator>
+#include <json/reader.h>
+#include <json/value.h>
+#include <string>
+
+namespace {
+Mantid::Kernel::Logger g_log("ISISInstrDataCache");
+} // namespace
+
+std::string Mantid::API::ISISInstrDataCache::getFileParentDirPath(std::string fileName) {
+  g_log.debug() << "ISISInstrDataCache::getFileParentDirPath(" << fileName << ")" << std::endl;
+
+  // Check if suffix eg. -add is present in filename
+  std::string suffix = FileFinder::Instance().extractAllowedSuffix(fileName);
+  if (!suffix.empty()) {
+    throw std::invalid_argument("Unsuported format: Suffix detected.");
+  }
+
+  // Find the last non-digit as the instrument name can contain numbers
+  std::string::reverse_iterator itRev = std::find_if(fileName.rbegin(), fileName.rend(), std::not_fn(isdigit));
+  std::string::size_type nChars = std::distance(itRev, fileName.rend());
+
+  // Check run number
+  std::string runNumber = fileName.substr(nChars);
+  if (runNumber.empty() || !std::all_of(runNumber.begin(), runNumber.end(), ::isdigit)) { // Check for wrong format
+    throw std::invalid_argument("Filename not in correct format.");
+  };
+  runNumber.erase(0, runNumber.find_first_not_of('0')); // Remove padding zeros
+
+  // Get instrument full name from short name
+  std::transform(fileName.begin(), fileName.end(), fileName.begin(), toupper);
+  std::string instrName = fileName.substr(0, nChars);
+  try {
+    auto instrInfo = FileFinder::Instance().getInstrument(instrName, true);
+    instrName = instrInfo.name();
```

Comment:
I don't think `instrName` needs to be assigned on line 42 if it will always get overwritten on line 45?